.github/actions/filter-files/action.yml

```diff
@@ -23,15 +23,15 @@ runs:
         script: |
           const extensionsRaw = "${{ inputs.extensions }}";
           const exactFilesRaw = "${{ inputs.files }}";
-          const inputFiles = "${{ inputs.changed-files }}".split(' ').filter(Boolean);
+          const inputFiles = JSON.parse(`${{ inputs.changed-files }}`);
           const extensions = extensionsRaw.trim() ? extensionsRaw.split(',') : [];
           const exactFiles = exactFilesRaw.trim() ? exactFilesRaw.split(',') : [];
 
           const result = inputFiles.filter(name => {
             return extensions.some(ext => name.endsWith(ext)) || (
               exactFiles.includes(name)
             )
-          }).join(' ');
-          console.log(`Filtered Files: ${result}`)
+          })
+          console.log(`Filtered Files: ${JSON.stringify(result)}`)
           return result;
```

Comment:
This should be `JSON.stringify()`'d yet, right, or does the `result-encoding` change on line 37 manage that?